sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -479,6 +484,21 @@ class depends_on {
 }
 ----
 
+==== Depends-On-all-leaves Property 
+The API for explicitly adding nodes to a `command_graph` includes a
+`property_list` parameter. This extension defines the `depends_on_all_leaves` 
+property to be passed here. `depends_on_all_leaves` provides a shortcut for 
+adding all leaves of a graph as dependencies.
+Therefore the created node will depend on all the current leaves in the graph.  
```

Comment:
Wondering whether we should add some sort of nod to barriers here, e.g "adding an empty node with this property will act as a barrier to future commands"